var/spack/repos/builtin/packages/file/package.py

```diff
@@ -38,6 +38,8 @@ class File(AutotoolsPackage):
     depends_on("zlib-api")
     depends_on("zstd", when="@5.44:")
 
+    provides("file")
```

Comment:
> I'm not actually sure what happens when a provider name matches a virtual name but we generally avoid it

This was also a concern I had but it didn't _seem_ to break anything?


So the reason I flipped to @alalazo's position on this PR was because I am pretty constantly thinking about how we can offer packages that don't build with Windows native tools (or require unix headers/libs), but can be compiled for the platform (Conan and Vcpkg both offer "unix only" packages via prebuilt binaries, similar to what we do here). Since we're already building these packages, it made sense to me that we offer them as first class packages (basically Max's suggestion). However, I felt having to differentiate "file" vs "win-file" across platforms/contexts really limited the utility of even offering `win-file` as a first class package beyond bootstrapping, as it requires uses to think about what platform they're on to select a package name, and makes envs less distributable/easily cross platform. Ideally the users would just request "file" and depending on the platform they're on, they would be served the correct package. AFAIK, virtuals are the only way to do this

Reply:
>> I'm not actually sure what happens when a provider name matches a virtual name but we generally avoid it

> This was also a concern I had but it didn't seem to break anything?

I think even if it works, it's a good policy to use separate names because otherwise the error messages can be confusing. If we wanted to go the route of allowing package names to shadow virtual names, I think we should also explicitly test for it (but overall I think that should be a separate PR, so if we keep a virtual, I think that `file` and `win-file` should both provide something new for now, like `file-util`).

> I felt having to differentiate "file" vs "win-file" across platforms/contexts really limited the utility of even offering win-file as a first class package beyond bootstrapping, 

True. However, this is an atypical usage of virtuals: typically multiple providers for a virtual would be expected to be interchangeable but as stated above, the `win-file` package would not be desired/usable on non-Windows (would it?); that makes me want to take a step back and consider how the model could accommodate users trying to install `file` functionality on Windows.

To clarify, this is not an issue of lacking constructs in Spack - you can say `if sys.platform == "win32": provides(...)` - but more about whether handling this use case by introducing a non-idiomatic provider is the way to go.